src/prefect/server/models/task_runs.py

```diff
@@ -51,34 +51,55 @@ async def create_task_run(
     now = pendulum.now("UTC")
 
     # if a dynamic key exists, we need to guard against conflicts
-    insert_stmt = (
-        (await db.insert(db.TaskRun))
-        .values(
+    if task_run.flow_run_id:
+        insert_stmt = (
+            (await db.insert(db.TaskRun))
+            .values(
+                created=now,
+                **task_run.dict(
+                    shallow=True, exclude={"state", "created"}, exclude_unset=True
+                ),
+            )
+            .on_conflict_do_nothing(
+                index_elements=db.task_run_unique_upsert_columns,
+            )
+        )
+        await session.execute(insert_stmt)
+
+        query = (
+            sa.select(db.TaskRun)
+            .where(
+                sa.and_(
+                    db.TaskRun.flow_run_id == task_run.flow_run_id,
+                    db.TaskRun.task_key == task_run.task_key,
+                    db.TaskRun.dynamic_key == task_run.dynamic_key,
+                )
+            )
+            .limit(1)
+            .execution_options(populate_existing=True)
+        )
+        result = await session.execute(query)
+        model = result.scalar()
+    else:
+        # Insert a new task run with a specific id
+        new_task_run_id = uuid4()
+        insert_stmt = (await db.insert(db.TaskRun)).values(
+            id=new_task_run_id,
             created=now,
             **task_run.dict(
                 shallow=True, exclude={"state", "created"}, exclude_unset=True
             ),
         )
-        .on_conflict_do_nothing(
-            index_elements=db.task_run_unique_upsert_columns,
-        )
-    )
-    await session.execute(insert_stmt)
-
-    query = (
-        sa.select(db.TaskRun)
-        .where(
-            sa.and_(
-                db.TaskRun.flow_run_id == task_run.flow_run_id,
-                db.TaskRun.task_key == task_run.task_key,
-                db.TaskRun.dynamic_key == task_run.dynamic_key,
-            )
+        await session.execute(insert_stmt)
+
+        query = (
+            sa.select(db.TaskRun)
+            .where(db.TaskRun.id == new_task_run_id)
+            .limit(1)
+            .execution_options(populate_existing=True)
         )
-        .limit(1)
-        .execution_options(populate_existing=True)
-    )
-    result = await session.execute(query)
-    model = result.scalar()
+        result = await session.execute(query)
+        model = result.scalar()
```

Comment:
Q: I'm curious why we're doing an insert followed immediately by a query to get the model from the DB. Is that a form of sanity checking/retrieving values set by the DB? It looks like this is pre-existing behavior so 🤞🏼 the answer isn't lost to time.